cypress/e2e/right-panel/pinned-messages-card.spec.ts

```diff
@@ -0,0 +1,202 @@
+        });
+
+        // Take a snapshot of RightPanel
+        cy.get(".mx_RightPanel").percySnapshotElement("Pinned messages card - pinned messages", {
+            percyCSS,
+            widths: [264], // Emulate the UI. The value is based on minWidth specified on MainSplit.tsx
+        });
+    });
+
+    it("should update when messages are unpinned", () => {
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findByRole("button", { name: "Pinned messages" }).should("be.visible").click();
+        });
+
+        sendAndPin("The second one");
+
+        cy.get(".mx_PinnedMessagesCard").within(() => {
+            cy.get(".mx_PinnedEventTile").should("have.length", 2);
+
+            // Unpin the first message
+            cy.get(".mx_PinnedEventTile").last().realHover().findByRole("button", { name: "Unpin" }).click();
+            cy.get(".mx_PinnedEventTile").should("have.length", 1);
+
+            // Unpin the second message
+            cy.get(".mx_PinnedEventTile").last().realHover().findByRole("button", { name: "Unpin" }).click();
+            cy.get(".mx_PinnedEventTile").should("have.length", 0);
+
+            // Wait until the information about the empty state is rendered
+            cy.get(".mx_PinnedMessagesCard_empty").should("exist");
+        });
+
+        // Take a snapshot of RightPanel
+        cy.get(".mx_RightPanel").percySnapshotElement("Pinned messages card - empty", {
+            percyCSS,
+            widths: [264], // Emulate the UI. The value is based on minWidth specified on MainSplit.tsx
+        });
+    });
+
+    it("should account for edits", () => {
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findByRole("button", { name: "Pinned messages" }).should("be.visible").click();
+        });
+
+        cy.get(".mx_RoomView_body").within(() => {
+            // Hover the event tile and click the edit button
+            cy.contains(".mx_EventTile", "First pinned message")
+                .realHover()
+                .findByRole("button", { name: "Edit" })
+                .click();
+
+            // Edit the message
+            cy.get(".mx_EditMessageComposer").findByRole("textbox").type(", edited{enter}");
+        });
+
+        cy.get(".mx_PinnedMessagesCard").within(() => {
+            cy.get(".mx_PinnedEventTile").should("have.length", 1);
+
+            // Assert that the edited pinned message is rendered
+            cy.get(".mx_PinnedEventTile .mx_EventTile_body").findByText("First pinned message, edited").should("exist");
+
+            // Assert that text link button is rendered
+            cy.findByRole("button", { name: /Edited/ }).should("exist");
+        });
+
+        // Take a snapshot of RightPanel
+        cy.get(".mx_RightPanel").percySnapshotElement("Pinned messages card - an edited message", {
+            percyCSS,
+            widths: [264], // Emulate the UI. The value is based on minWidth specified on MainSplit.tsx
+        });
```

Comment:
I wonder if we really need so many snapshots. We only have a limited number of Percy snapshots to use every month, so we should be mindful about how best to use them. I think maybe we could get away with one or two for this file?